### ISSUE TITLE
Analytics additions: tracking login step successes/fails

### ISSUE THREAD

**Issue by g-opzt** [department-of-veterans-affairs/va-mobile-app#6228]
## ACs
Add the following events. 

- [ ] `vama_user_call`: Return the response code of the API call to the user endpoint in a param outcome
     - [ ] `status_code`: status code from the API return

- [ ] `vama_login_token_get`: Returns the response code of the request for access and refresh tokens on /token endpoint
    - [ ] `status_code`: status code returned from the /token endpoint

- [ ] `vama_login_token_store`: Return whether or not the app's attempt to store tokens in app storage succeeded
     - [ ] `success`: true if tokens stored (should be in the try statement), false if not (in the catch statement)

- [ ] `vama_login_token_retrieve (check length)`: Return whether or not the the app's attempt to retrieve tokens in app storage
     - [ ] `success`: true if token retrieved, false if not

- [ ] `vama_login_token_refresh`: Return the response code of the call to the /refresh endpoint
     - [ ] `status_code`: status code returned from the /refresh endpoint

**Comment by TKDickson** [comment]
@g-opzt @ajsarkar28 what's the relative timeline for this work? Super roughly, like "this quarter" or "not yet prioritized/dunno"

**Comment by g-opzt** [comment]
@TKDickson This is primarily a blocker to refining our retention understanding but there's no specific work planned, that I know of, to actually dig in on login issues from our team's side currently. It is blocking the dashboard for Login which was the top of Rachel's priority list for dashboards--but we also don't have any points planned for dashboard work this quarter. @ajsarkar28 this would be a good thing to surface with PMs/leadership for planning.

**Comment by TKDickson** [comment]
@g-opzt thank you for the clarification! 

I know I'm preaching to the choir here, and want to have this on the ticket somewhere -- we still see login-related bug reports coming in from users that we can't replicate internally, so having more clear analytics (events firing to better track what's going on, how often this is occurring, etc) we're hoping could give us better insight on when and why those are happening (are the biometrics problems mostly because they're logging in less frequently than every 45 days, or something else?? etc)

**Comment by theodur** [comment]
The `vama_login_token_retrieve` event name is 1 character over the 24-character limit, so I had to shuffle some things around with the event names. I ended up renaming `vama_login_token_retrieve` to `vama_login_token_get` and renaming the original `vama_login_token_get` event to `vama_login_token_fetch`. 

Appending `fetch` to the event name responsible for tracking the status codes from the `/token` should help signify that it's tracking fetching from the API rather than getting the token from async storage. Likewise with the `vama_login_token_get` event and async storage.
